### ISSUE TITLE
[Bug]: Observe Crash on OneSignal.initWithContext(getRepository().getContext(),oneSignalId) with Version 5.0.0

### ISSUE THREAD

**Issue by suyashraikar** [OneSignal/OneSignal-Android-SDK#1813]
### What happened?

The latest (5.0.0) version of SDK crashes during initialization using OneSignal.initWithContext(Context, String) method
The attached crash log is extracted from my application

### Steps to reproduce?

```Markdown
- Install Release 5.0.0 version
- Initalize SDK using OneSignal.initWithContext(getRepository().getContext(),oneSignalId) in Application class
- Observe crash
```

### What did you expect to happen?

Code should work without any crash 

### OneSignal Android SDK version

Release 5.0.0

### Android version

13

### Relevant log output

```Shell
Process: com.simpletherapy.android.mhealth, PID: 18149
                                                                                                    java.lang.RuntimeException: Unable to create application com.simpletherapy.android.SimpleTherapyApplication: java.lang.NullPointerException: null cannot be cast to non-null type com.amazon.device.iap.PurchasingListener
                                                                                                    	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6955)
                                                                                                    	at android.app.ActivityThread.access$1300(ActivityThread.java:260)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1972)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loop(Looper.java:223)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7989)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:635)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1065)
                                                                                                    Caused by: java.lang.NullPointerException: null cannot be cast to non-null type com.amazon.device.iap.PurchasingListener
                                                                                                    	at com.onesignal.core.internal.purchases.impl.TrackAmazonPurchase.start(TrackAmazonPurchase.kt:83)
                                                                                                    	at com.onesignal.core.internal.startup.StartupService.start(StartupService.kt:16)
                                                                                                    	at com.onesignal.internal.OneSignalImp.initWithContext(OneSignalImp.kt:247)
                                                                                                    	at com.onesignal.OneSignal.initWithContext(OneSignal.kt:126)
                                                                                                    	at com.simpletherapy.android.SimpleTherapyApplication.onCreate(SimpleTherapyApplication.java:187)
                                                                                                    	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1192)
                                                                                                    	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6949)
                                                                                                    	at android.app.ActivityThread.access$1300(ActivityThread.java:260) 
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1972) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                                                    	at android.os.Looper.loop(Looper.java:223) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7989) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:635) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1065)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jennantilla** [comment]
Hi @suyashraikar thanks for reaching out! We'll investigate this.

**Comment by suyashraikar** [comment]
@jennantilla I am using OneSignal SDK [5.0.0, 5.99.99] 

minSdkVersion    : 21, // 19 kitkat,
targetSdkVersion : 34,
compileSdkVersion: 34



Also enable required details for Amazon app store too.

**Comment by suhaspalkar** [comment]
// 2.0.1
            val listenerHandlerClass = Class.forName("com.amazon.device.iap.internal.d")
            listenerHandlerObject = try {
                // iap v2.x
                listenerHandlerClass.getMethod("d").invoke(null)
            } catch (e: NullPointerException) {
                // appstore v3.x
                listenerHandlerClass.getMethod("e").invoke(null)
                registerListenerOnMainThread = true
            }
            val locListenerHandlerField = listenerHandlerClass.getDeclaredField("f")
            locListenerHandlerField.isAccessible = true
            osPurchasingListener = OSPurchasingListener(_operationRepo, _configModelStore, _identityModelStore)
            osPurchasingListener!!.orgPurchasingListener = **locListenerHandlerField.get(listenerHandlerObject) as PurchasingListener**

            listenerHandlerField = locListenerHandlerField

**Comment by suhaspalkar** [comment]
Class name: TrackAmazonPurchase
Line No. : #83
osPurchasingListener!!.orgPurchasingListener = locListenerHandlerField.get(listenerHandlerObject) as PurchasingListener
@jennantilla This line is causing for the crash because 'locListenerHandlerField.get(listenerHandlerObject)' is null.

**Comment by suyashraikar** [comment]
@jennantilla  any updates ?

**Comment by arthurgeron-work** [comment]
Unfortunately the temporary fix for me was to add try/catch, has happened only on Xiaomi or Motorola devices on Android 13+.
I have two separate stack traces, all from initWithContext:

```
Exception java.lang.RuntimeException:
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:7619)
  at android.app.ActivityThread.-$$Nest$mhandleBindApplication
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2400)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loopOnce (Looper.java:226)
  at android.os.Looper.loop (Looper.java:313)
  at android.app.ActivityThread.main (ActivityThread.java:8757)
  at java.lang.reflect.Method.invoke
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:571)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1067)
Caused by java.lang.NullPointerException:
  at com.onesignal.core.internal.application.impl.ApplicationService.getAppContext (ApplicationService.kt:3)
  at com.onesignal.core.internal.device.impl.DeviceService.packageInstalledAndEnabled (DeviceService.kt:3)
  at com.onesignal.core.internal.device.impl.DeviceService.isGMSInstalledAndEnabled (DeviceService.kt:3)
  at com.onesignal.core.internal.device.impl.DeviceService.supportsGooglePush (DeviceService.kt:9)
  at com.onesignal.core.internal.device.impl.DeviceService.getDeviceType (DeviceService.kt:10)
  at com.onesignal.core.internal.device.impl.DeviceService.isAndroidDeviceType (DeviceService.kt:1)
  at com.onesignal.location.LocationModule$register$1.invoke (LocationModule.kt:3)
  at com.onesignal.location.LocationModule$register$1.invoke (LocationModule.kt:1)
  at com.onesignal.common.services.ServiceRegistrationLambda.resolve (ServiceRegistration.kt:13)
  at com.onesignal.common.services.ServiceProvider.getServiceOrNull (ServiceProvider.kt:51)
  at com.onesignal.common.services.ServiceProvider.getService (ServiceProvider.kt:6)
  at com.onesignal.common.services.ServiceRegistrationReflection.resolve (ServiceRegistration.kt:208)
  at com.onesignal.common.services.ServiceProvider.getServiceOrNull (ServiceProvider.kt:51)
  at com.onesignal.common.services.ServiceProvider.getService (ServiceProvider.kt:6)
  at com.onesignal.common.services.ServiceRegistrationReflection.resolve (ServiceRegistration.kt:208)
  at com.onesignal.common.services.ServiceProvider.getServiceOrNull (ServiceProvider.kt:51)
  at com.onesignal.common.services.ServiceProvider.getService (ServiceProvider.kt:6)
  at com.onesignal.internal.OneSignalImp.initWithContext (OneSignalImp.kt:233)
  at com.onesignal.OneSignal.initWithContext (OneSignal.kt:1)
  at com.tradersclub.tc.MainApplication.onCreate (MainApplication.java:25)
  at android.app.Instrumentation.callApplicationOnCreate (Instrumentation.java:1266)
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:7614)
  ```
  
  Second:
   ```
   
First stacktrace:

Exception java.lang.RuntimeException:
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:7232)
  at android.app.ActivityThread.-$$Nest$mhandleBindApplication
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2241)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loopOnce (Looper.java:211)
  at android.os.Looper.loop (Looper.java:300)
  at android.app.ActivityThread.main (ActivityThread.java:8395)
  at java.lang.reflect.Method.invoke
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:559)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:954)
Caused by java.lang.reflect.InvocationTargetException:
  at java.lang.reflect.Constructor.newInstance0
  at java.lang.reflect.Constructor.newInstance (Constructor.java:343)
  at com.onesignal.common.services.ServiceRegistrationReflection.resolve (ServiceRegistration.kt:238)
  at com.onesignal.common.services.ServiceProvider.getServiceOrNull (ServiceProvider.kt:51)
  at com.onesignal.common.services.ServiceProvider.getService (ServiceProvider.kt:6)
  at com.onesignal.internal.OneSignalImp.initWithContext (OneSignalImp.kt:76)
  at com.onesignal.OneSignal.initWithContext (OneSignal.kt:1)
  at com.tradersclub.tc.MainApplication.onCreate (MainApplication.java:25)
  at android.app.Instrumentation.callApplicationOnCreate (Instrumentation.java:1289)
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:7226)
Caused by java.lang.NullPointerException:
  at com.onesignal.core.internal.application.impl.ApplicationService.getAppContext (ApplicationService.kt:3)
  at com.onesignal.core.internal.preferences.impl.PreferencesService.getSharedPrefsByName (PreferencesService.kt:4)
  at com.onesignal.core.internal.preferences.impl.PreferencesService.get (PreferencesService.kt:38)
  at com.onesignal.core.internal.preferences.impl.PreferencesService.getString (PreferencesService.kt:13)
  at com.onesignal.common.modeling.ModelStore.load (ModelStore.kt:29)
  at com.onesignal.common.modeling.SimpleModelStore.<init> (SimpleModelStore.kt:4)
  at com.onesignal.core.internal.config.ConfigModelStore.<init> (ConfigModelStore.kt:12)
```

**Comment by jennantilla** [comment]
Hello everyone, apologies that I did not follow up here. This should be fixed in SDK versions [5.0.3](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/5.0.3) and later. Please let us know if you are still experiencing any issues after upgrading.

Thanks!
